internal/dataplane/parser/translators/httproute.go

```diff
@@ -350,14 +352,46 @@ func mustMarshalJSON[T any](val T) string {
 	return string(key)
 }
 
-// GeneratePluginsFromHTTPRouteFilters converts HTTPRouteFilter into Kong plugins.
+// ConvertFiltersToPlugins converts HTTPRouteFilter into Kong plugins.
+// The plugins can be set in two different ways:
+// - Direct conversion from the respective HTTPRouteFilter.
+// - ExtensionRef to plugins annotation from the ExtensionRef filter.
```

Comment:
We should mention here that the `route` is going to be updated in place.

Also the name suggests that there's going to be an output (you usually convert something into something).

Reply:
I agree, I changed the function name and improved the godoc section.